sql/catalyst/src/test/scala/org/apache/spark/sql/types/DataTypeWriteCompatibilitySuite.scala

```diff
@@ -510,6 +510,110 @@ abstract class DataTypeWriteCompatibilityBaseSuite extends SparkFunSuite {
       "Should allow map of int written to map of long column")
   }
 
+  test("Check udt: underlying sql type is same") {
```

Comment:
If you don't mind, could you add a test name prefix, `SPARK-47528: `, for the newly added three test cases?

Reply:
Added the prefix.